### DISCUSSION TITLE
is not a module.

### DISCUSSION THREAD

**Discussion by manikerisaurabh** [vercel/next.js#73219] [Help]
### Summary

I'am building my next js 15 app and whiling building i got an error like is not a module
![image](https://github.com/user-attachments/assets/75e1b498-cf20-489b-82f5-a6b53ab3d73f)

I resolved this error by making sure that is there any empty file is created or not I see what i found that in my /write-testimonial route there is a file named as page.tsx which i had created but did not written anything so i deleted that file and tried building after that and i was impressed that my build was ready to server....

**Comment by icyJoseph** [comment]
Do you mean, that you had created the `/write-testimonial/page.tsx` file, but it was empty, and therefore you couldn't build?

That makes sense. `page.js` (or `.tsx`) is sort of the semantic entry point for page view, to simplify a lot, and if its empty, then it has not exports/imports, which makes it `not-a-module` on the eyes of TypeScript.

**Reply by manikerisaurabh** [reply]
Yes, that’s correct. I had created the `/write-testimonial/page.tsx` file but left it completely empty, without any exports or content. During the build process, this caused an issue where TypeScript flagged it as "not a module." Since the file did not contain any valid module exports or imports, it essentially became unusable in the eyes of the compiler.

Deleting this empty file resolved the issue, and the build completed successfully. It's a good reminder to ensure that all ` page.tsx  ` file have valid content or are removed if not being utilized. Thanks for pointing that out!

**Reply by icyJoseph** [reply]
It goes beyond that. I explored that in this answer, and follow ups, to a discussion, nearly two years ago:

- https://github.com/vercel/next.js/discussions/50684#discussioncomment-6071330

**Reply by manikerisaurabh** [reply]
Thank you for the additional context! I appreciate you pointing that out. I hadn’t come across that discussion, and it’s helpful to know that this issue has been discussed before. I’ll make sure to check those resources in the future to avoid missing out on useful insights. Thanks again for sharing your experience!
